src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/ClientServices.cs

```diff
@@ -45,14 +45,17 @@ public class ClientServices : IClientHost, IEndpointServices, IEndpointDiscovery
 
                 // copy the certificate, public key only, into the trusted certificates list
                 try {
-                    var publicKey = new X509Certificate2(value.RawData);
+
+
```

Comment:
Maybe consider moving the entire "setter" into a private method now that it is becoming quite complex